erpnext/regional/doctype/import_supplier_invoice/import_supplier_invoice.py

```diff
@@ -0,0 +1,396 @@
+
+		if args.city:
+			make_address.city = args.city
+		else:
+			make_address.city = "Not Provided"
+
+		if args.province:
+			make_address.state_code = args.province
+
+		if args.pincode:
+			make_address.pincode = args.pincode
+
+		if args.country:
+			make_address.country = args.country
+
+		for address in existing_address:
+			address_doc = frappe.get_doc("Address", address["name"])
+			if (address_doc.address_line1 == make_address.address_line1 and
+				address_doc.pincode == make_address.pincode):
+				return address
+
+		make_address.append("links", {
+			"link_doctype": "Supplier",
+			"link_name": args.supplier_name
+		})
+		make_address.address_type = "Billing"
+		make_address.insert()
+		return make_address.name
+	else:
+		return None
+
+def create_purchase_invoice(**args):
+
+	args = frappe._dict(args)
+	pi = frappe.get_doc({
+			"doctype": "Purchase Invoice",
+			"company": args.company,
+			"naming_series": args.naming_series,
+			"supplier": args.supplier_name,
+			"is_return": args.is_return,
+			"posting_date": today(),
+			"bill_no": args.bill_no,
+			"bill_date": args.bill_date,
+			"destination_code": args.destination_code,
+			"document_type": args.document_type,
+			"items": args["items"],
+			"taxes": args["taxes"]
+		})
+
+	try:
+		pi.insert(ignore_permissions=True)
+
+		if args.total_discount > 0:
+			pi.apply_discount_on = "Grand Total"
+			pi.discount_amount = args.total_discount
+			pi.save()
+
+		calc_total = 0
+		adj = 0
+		for term in args.terms:
+			calc_total += float(term["payment_amount"])
+		if float(calc_total - float(pi.grand_total)) != 0:
+			adj = calc_total - float(pi.grand_total)
+		pi.payment_schedule = []
+		for term in args.terms:
+			pi.append('payment_schedule',{"mode_of_payment_code": term["mode_of_payment_code"],
+			"bank_account_iban": term["bank_account_iban"],
+			"due_date": term["due_date"],
+			"payment_amount": float(term["payment_amount"]) - adj })
+			adj = 0
+		pi.imported_grand_total = calc_total
+		pi.save()
+		return pi.name
+	except Exception as e:
+		frappe.log_error(message=e, title="Create Purchase Invoice: " + args.bill_no + "File Name: " + args.file_name)
+		return None
+
+def get_country(code):
+
+	existing_country_name = frappe.db.get_value("Country",
+			filters={"code": code}, fieldname="name")
+	if existing_country_name:
+		return existing_country_name
+	else:
+		frappe.throw(_("Country Code in File does not match with country code set up in the system"))
+
+def create_uom(uom):
+
+	existing_uom = frappe.db.get_value("UOM",
+			filters={"uom_name": uom}, fieldname="uom_name")
+	if existing_uom:
+		return existing_uom
+	else:
+		new_uom = frappe.new_doc("UOM")
+		new_uom.uom_name = uom
+		new_uom.save()
+		return new_uom.uom_name
+
+def check_bill_no(invoice_no):
+
+	existing_bill_no = frappe.db.get_value("Purchase Invoice",
+				filters={"bill_no": invoice_no, "docstatus": 1}, fieldname="name")
+	if existing_bill_no:
+		return existing_bill_no
+	else:
+		return None
```

Comment:
No need of return None, by default function returns None

Reply:
removed this method, it is not needed as duplicate supplier invoice check is handled through configuration at "Accounts settings" level